src/main/java/org/wso2/carbon/connector/CreateJWT.java

```diff
@@ -0,0 +1,182 @@
+        signature.update(data);
+        return signature.sign();
+    }
+
+    /**
+     * The method is using to extract the private key from the p12 file
+     *
+     * @param keyFile the p12 file to extract the private key
+     * @param password the password to extract the p12 file
+     * @return the private key
+     * @throws KeyStoreException
+     * @throws IOException
+     * @throws NoSuchAlgorithmException
+     * @throws CertificateException
+     * @throws UnrecoverableKeyException
+     */
+    private static PrivateKey getPrivateKey(String keyFile, String password)
+            throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException,
+            UnrecoverableKeyException {
+
+        KeyStore keystore = KeyStore.getInstance(JWTConstant.KEY_STORE);
+        keystore.load(new FileInputStream(keyFile), password.toCharArray());
+        return (PrivateKey) keystore.getKey(keyAlias, password.toCharArray());
+    }
+
+    /**
+     * The method is using to construct the Json Web Token
+     * @param messageContext the message context
+     * @return the json web token
+     * @throws InvalidKeyException
+     * @throws SignatureException
+     * @throws NoSuchAlgorithmException
+     * @throws UnrecoverableKeyException
+     * @throws KeyStoreException
+     * @throws CertificateException
+     * @throws IOException
+     */
+    private static String getJsonWebToken(MessageContext messageContext) throws
+            InvalidKeyException, SignatureException, NoSuchAlgorithmException,
+            UnrecoverableKeyException, KeyStoreException, CertificateException, IOException {
+
+        String keyStoreLocation = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                JWTConstant.KEY_STORE_LOCATION);
+        String serviceAccount = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                JWTConstant.SERVICE_ACCOUNT);
+        String scope = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                JWTConstant.SCOPE);
+        String password = JWTConstant.PASSWORD;
+        String jwtHeaderStr;
+        String jwtClaimStr;
+        PrivateKey privateKey;
+
+        //Construct JWT Header
+        JSONObject jwtHeader = new JSONObject();
+        try {
+            jwtHeader.put(JWTConstant.JWT_HEADER_ALGO, JWTConstant.JWT_HEADER_ALGO_VALUE);
+            jwtHeader.put(JWTConstant.JWT_HEADER_TYPE, JWTConstant.JWT_HEADER_TYPE_VALUE);
+            jwtHeaderStr = jwtHeader.toString();
+        } catch (JSONException e) {
+            throw new SynapseException(e.getMessage(), e);
+        }
+
+        byte[] encodedHeader = new byte[0];
+        if (jwtHeaderStr != null) {
+            encodedHeader = Base64.encodeBase64(jwtHeaderStr.getBytes(JWTConstant.UTF));
+        }
+
+        //Construct JWT Claim
+        JSONObject jwtClaimSet = new JSONObject();
+        long iat = (System.currentTimeMillis() / 1000) - 60;
```

Comment:
Fixed in https://github.com/wso2-extensions/esb-connector-bigquery/pull/11/commits/074492ecf2bf9deab34a477ce509b792eb891ca3